ndc-lambda-sdk/src/execution.ts

```diff
@@ -211,20 +202,76 @@ function buildCausalStackTrace(error: Error): string {
   return stackTrace;
 }
 
-export function reshapeResultToNdcResponseValue(value: unknown, type: schema.TypeDefinition, valuePath: string[], fields: Record<string, sdk.Field> | "AllColumns", objectTypes: schema.ObjectTypeDefinitions): unknown {
+// Represents either selecting a scalar (ie. the whole value, opaquely), an object (selecting properties), or an array (select whole array)
+export type FieldSelection = sdk.NestedField | { type: "scalar" }
+
+function reshapeResultUsingFunctionCallingConvention(functionResultValue: unknown, functionResultType: schema.TypeDefinition, query: sdk.Query, objectTypes: schema.ObjectTypeDefinitions): sdk.RowSet {
+  if (query.aggregates) throw new sdk.NotSupported("Query aggregates are not supported");
+  if (query.order_by) throw new sdk.NotSupported("Query order_by is not supported");
+  if (query.predicate) throw new sdk.NotSupported("Query predicate is not supported");
+  if (!query.fields) {
+    return {
+      aggregates: null,
+      rows: null,
+    }
```

Comment:
Am I missing what happens to the result here?

Reply:
We throw them away because they selected null fields. In practice this won't happen (GraphQL forces you to select _something_), but the spec (Phil) says that this is the correct behaviour (return null rows if fields is omitted)